### PR TITLE
Add Info for cache tags

### PR DESCRIPTION
**Pull request by nikischin**
PR for #3710

### CODE REVIEW

**Comment by phryneas** [comment]
I'm pretty sure that's incorrect - the plain string array would be considered in success and error cases, and you need the function if you want those to return different values depending on success or not.

**Comment by nikischin** [comment]
> I'm pretty sure that's incorrect - the plain string array would be considered in success and error cases, and you need the function if you want those to return different values depending on success or not.

In #3710 I experienced differently. However, I am happy to be proofed wrong. I wasn't able to find this part in the code, do you know which part of the code is responsible for it?

**Comment by phryneas** [comment]
That would be 

https://github.com/reduxjs/redux-toolkit/blob/9c0d553c574691244e25ddcc262170e8744883b1/packages/toolkit/src/query/core/buildThunks.ts#L662-L678

 and 

https://github.com/reduxjs/redux-toolkit/blob/9c0d553c574691244e25ddcc262170e8744883b1/packages/toolkit/src/query/endpointDefinitions.ts

`providesTags` will only not behave like this if your queryFn throws an error instead of returning `{ error: ... }`

File: docs/rtk-query/usage/automated-refetching.mdx [hunk 1]
```diff
@@ -38,6 +38,8 @@ A _query_ can have its cached data _provide_ tags. Doing so determines which 'ta
 
 The `providesTags` argument can either be an array of `string` (such as `['Post']`), `{type: string, id?: string|number}` (such as `[{type: 'Post', id: 1}]`), or a callback that returns such an array. That function will be passed the result as the first argument, the response error as the second argument, and the argument originally passed into the `query` method as the third argument. Note that either the result or error arguments may be undefined based on whether the query was successful or not.
 
+Please note, tags provided in an simple array (such as `['Post', 'User']`) and not a callback are only considered for successful requests. In order to provide tags also for failed requests the callback has to be used. With the callback it is also possible providing different tags for failed or successful requests.
```
Comment (markerikson) [inline docs/rtk-query/usage/automated-refetching.mdx:41]:
```suggestion

Note that tags provided in an simple array (such as `['Post', 'User']`) and not a callback are only considered for successful requests. In order to provide tags for failed requests the callback has to be used. With the callback it is also possible to provide different tags for failed or successful requests.
```

**Comment by markerikson** [comment]
Hmm. Sounds like there's some confusion going on here, actually.

**Comment by EskiMojo14** [comment]
here's a quick sandbox showing that failed queries do provide tags when using the array form for providesTags: https://codesandbox.io/p/sandbox/failed-query-providing-tags-t4yjv7

**Pull request closed without merging**